test/test-shim.ts

```diff
@@ -70,6 +74,37 @@ test('modules already in cached do not show up in trace', t => {
   t.end();
 });
 
+test('preload traces from the entrypoint and writes it to an env controlled file', t => {
+  const script = join(__dirname, './fixtures/modA.js');
+  const rssPath = join(__dirname, '../src/index.js');
+  const tracePath = join(tmpdir(), `./require-so-slow.trace`);
+  const command = `TRACE_OUTFILE=${tracePath} node -r ${rssPath} ${script}`;
+  execSync(command);
+  t.true(existsSync(tracePath));
+  const events: Array<{ name: string }> = JSON.parse(
+    readFileSync(tracePath, 'utf8')
+  );
+  unlinkSync(tracePath);
+  // Can't test that 'require /.../index.js' is the first event because nyc
+  // hacks things into the node subprocess
+  t.assert(events.find(e => e.name === `require ${resolve(script)}`));
+  t.end();
+});
+
+test('preload writes to require-so-slow.trace by default', t => {
+  const script = join(__dirname, './fixtures/modA.js');
+  const rssPath = join(__dirname, '../src/index.js');
+  const prevDir = process.cwd();
+  process.chdir(tmpdir());
```

Comment:
This logic is repeated between the tests. You can probably put this in `beforeEach` section (with a corresponding `afterEach` section restoring the directory). This would make the test logic more understandable by removing the boilerplate. It also makes the tests less interfering (an error in one of the tests is less likely to mess up the subsequent ones).

Reply:
It looks like `tape` doesn't include `beforeEach` and `afterEach` facilities. Does it make sense to extract the shared work into a function?
```ts
function moveToEmptyTempDir(): string {
  const dir = mkdtempSync('rss-test');
  const prev = process.cwd();
  process.chdir(dir);
  return prev;
}
```
A test then looks like:
```ts
test('something with file output', t => {
  const prevDir = moveToEmptyTempDir();
  ...
  process.chdir(prevDir);
  t.end();
});
```

Reply:
sgtm!